src/shared/platform/lind_platform.c

```diff
@@ -790,3 +780,25 @@ int lind_socket (int domain, int type, int protocol, int cageid)
     LIND_API_PART2;
     LIND_API_PART3;
 }
+
+int lind_getsockname (int sockfd, struct sockaddr * addr, socklen_t* addrlen, int cageid)
+{
+    LIND_API_PART1;
+    callArgs = Py_BuildValue("(i[iii])", LIND_safe_net_getsockname, sockfd, *addrlen, cageid);
+    LIND_API_PART2;
+    if (*addrlen > _len) {COPY_DATA(addr, _len);}
+    else {COPY_DATA(addr, _len);}
```

Comment:
Also the else case should be *addrlen, not _len, same for getpeername

Reply:
Fixed.